circuits/cpp/src/aztec3/circuits/kernel/private/common.cpp

```diff
@@ -61,15 +59,14 @@ void common_validate_call_stack(DummyBuilder& builder, PrivateCallData<NT> const
  * - https://discourse.aztec.network/t/spending-notes-which-havent-yet-been-inserted/180
  *
  * @param builder
- * @param storage_contract_address Contract address to use when siloing read requests
  * @param historic_private_data_tree_root This is a reference to the historic root which all
  * read requests are checked against here.
- * @param read_requests the commitments being read by this private call
+ * @param read_requests the commitments being read by this private call - transient reads here are
+ * `inner_note_hashes` (not yet siloed, not unique), but non-transient reads are `unique_siloed_note_hashes`
  * @param read_request_membership_witnesses used to compute the private data root
  * for a given request which is essentially a membership check
  */
 void common_validate_read_requests(DummyBuilder& builder,
-                                   NT::fr const& storage_contract_address,
                                    NT::fr const& historic_private_data_tree_root,
                                    std::array<fr, MAX_READ_REQUESTS_PER_CALL> const& read_requests,
                                    std::array<ReadRequestMembershipWitness<NT, PRIVATE_DATA_TREE_HEIGHT>,
```

Comment:
For a future PR: it might still be good to separate read requests from their membership witnesses.

Reply:
They are in separate arrays, but the membershipwitness info for readrequests is coupled with other info like `is_transient`.

Reply:
And there is this ticket to ensure that we stop forwarding `read_request_membership_witnesses` between kernels: https://github.com/AztecProtocol/aztec-packages/issues/1407